haystack/document_stores/pinecone.py

```diff
@@ -560,7 +560,7 @@ def delete_documents(
         index = self._sanitize_index_name(index)
         if index in self.pinecone_indexes:
             if ids is None and filters is None:
-                self.pinecone_indexes[index].delete(delete_all=True)
+                self.pinecone_indexes[index].delete()
```

Comment:
Your mock did not accept this parameter. As I imagine the mock represents the actual contract with Pinecone, I assumed this might be a leftover from older versions and removed it. 

Is that true? If not, I will revert this change and add `delete_all` to the mock.

Reply:
`delete_all` should be included in the mock